### PR TITLE
Vet clinic database: query multiple tables

### PR DESCRIPTION
**Pull request by KingsleyIbe**
# Milestone changes

In this milestone, I implemented the following:
- Created a table named owners with the provided columns

- Created a table named species with the provided columns

- Modified animals table as follows
      - Make sure that id is set as autoincremented PRIMARY KEY
      - Remove column species
      - Add column species_id which is a foreign key referencing species table
      - Add column owner_id which is a foreign key referencing the owners table
- Updated changes in the schema.sql file.
- Inserted the necessary data into the owners table as provided

- Inserted the necessary data into the species table as provided

- Modified the inserted animals so it includes the species_id value:
        - If the name ends in "mon" it will be Digimon
        - All other animals are Pokemon
- Modified the inserted animals to include owner information (owner_id) as provided

- Updated these insertions and modifications in the data.sql file.
- Wrote queries (using JOIN) to answer the following questions:
        - What animals belong to Melody Pond?
        - List of all animals that are pokemon (their type is Pokemon).
        - List all owners and their animals, remember to include those that don't own any animal.
        - How many animals are there per species?
        - List all Digimon owned by Jennifer Orwell.
        - List all animals owned by Dean Winchester that haven't tried to escape.
        - Who owns the most animals?

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -12,3 +14,42 @@ weight_kg       REAL       NOT NULL
...
+
+/* ADD COLUMN 'species_id' WHICH IS A FOREIGN KEY REFERENCING 'species' TABLE */
+ALTER TABLE animals
+  ADD species_id INT,
+  ADD CONSTRAINT fk_species
+  FOREIGN KEY (species_id)
+  REFERENCES species (id);
+
+/* ADD COLUMN 'owners_id' WHICH IS A FOREIGN KEY REFERENCING 'owners' TABLE */
+ALTER TABLE owners
```
Comment (Meltrust) [inline schema.sql]: - [ ] The table alteration for this change, should be on animals table (so it has a reference to owners table).

Example:

`   ALTER TABLE animals ADD owner_id int REFERENCES owners(id);
`

Please, kindly implement this correction 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @KingsleyIbe,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice code organization ✔️
- Nice use of queries ✔️ 
- Nice use of transactions ✔️  

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KingsleyIbe** [comment]
Hello @Meltrust 

Thank you for reviewing my project, be sure changes will be effected.

**Review by DammyShittu — APPROVED** [review]
Hi @KingsleyIbe, 

## APPROVED 🟢 

Great implementation of all the requirements in this milestone and the changes requested by the previous reviewer. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KingsleyIbe** [comment]
Hello @DammyShittu 

Thanks for reviewing and approving my work.

**Pull request merged**